aten/src/ATen/native/Pow.cpp

```diff
@@ -31,11 +31,13 @@ Tensor& pow_out(Tensor& result, const Tensor& base, Scalar exp) {
            "result type ", common_dtype, "can't be cast to the desired output type ",
            result.scalar_type());
 
-  auto exponent = (exp.isComplex()) ? exp.toComplexDouble() : exp.toDouble();
-
-  if (exponent == 0.0) {
+  if (exp.isComplex() && (exp.toComplexDouble() == 0.0) ) {
+    result.resize_as_(base).fill_(1);
+  } else if (exp.isComplex() && (exp.toComplexDouble() == 1.0) ) {
+    result.resize_as_(base).copy_(base);
+  } else if (!exp.isComplex() && (exp.toDouble() == 0.0)) {
     result.resize_as_(base).fill_(1);
-  } else if (exponent == 1.0) {
+  } else if (!exp.isComplex() && (exp.toDouble() == 1.0)) {
     result.resize_as_(base).copy_(base);
   } else {
```

Comment:
Can this just be:
```
auto exponent = exp.toComplexDouble();
if (exponent == 0.0) {
...
} else if (exponent == 1.0) {
...
} else {
...
}
```?
`exponent` doesn't actually participate in the computation so it doesn't affect the type promotion here (but maybe we should leave a comment here because it's very easy to make a mistake here)

Reply:
this is resolved with the `equal` function now